src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -842,10 +842,10 @@ static void range_check(MacroAssembler* masm, Register pc_reg, Register temp_reg
                         address code_start, address code_end,
                         Label& L_ok) {
   Label L_fail;
-  __ lea(temp_reg, ExternalAddress(code_start));
+  __ lea(temp_reg, AddressLiteral(code_start, relocInfo::none));
```

Comment:
What about when try to save and restore an adapter in Leyden?

Reply:
Short answer: we should exclude these checks when we cache adapters in Leyden.

Long answer:
1. These checks are enabled only in debug VM and only on x86.
2. We never caught any issues with these checks since JDK 9 (I found only one issue [JDK-8023465](https://bugs.openjdk.org/browse/JDK-8023465)).
3. Enabling it create issues because we don't add an adapters to hash table until all stubs are generated [contains_all_checks](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/sharedRuntime.cpp#L2637)  which which creates duplicated adapters for the same method's signature.
4. These checks may miss an issue because they pass if adapter called from stubs but stubs may be called from compiled call.

I was actually considering removing these checks but decided to keep them for now. But I don't think we need them in cached adapters in Leyden.

Reply:
I'm fine with that! (it makes save and restore in Leyden much simpler :-)